src/platform/tls_schannel.c

```diff
@@ -2170,6 +2255,21 @@ CxPlatTlsWriteDataToSchannel(
                     CertFreeCertificateContext(PeerCert);
                 }
 #endif
+            } else if (TlsContext->SecConfig->Flags & QUIC_CREDENTIAL_FLAG_REQUIRE_CLIENT_AUTHENTICATION
+                && CertValidationResult.hrVerifyChainStatus != QUIC_STATUS_SUCCESS) {
+                //
+                // If the server has configured client authentication but not deferred validation,
+                // fail the handshake if the client cert doesn't pass validation
+                //
```

Comment:
Your `else if` statement isn't correct. It's after a check for `QUIC_CREDENTIAL_FLAG_INDICATE_CERTIFICATE_RECEIVED` not `QUIC_CREDENTIAL_FLAG_DEFER_CERTIFICATE_VALIDATION`. This also leads me to a question:

If the peer doesn't defer certificate validation, but does set the indicate cert recv AND the inbox/default validation fails, should we even indicate the cert? In other words, should "indicate cert recv" == "indicate accepted cert recv" or should we indicate it regardless? I kind of lean towards only indicating it if things succeed.